src/librustc_errors/styled_buffer.rs

```diff
@@ -27,10 +27,21 @@ impl StyledBuffer {
     }
 
     fn replace_tabs(&mut self) {
-        for line in self.text.iter_mut() {
-            for c in line.iter_mut() {
+        for (line_pos, line) in self.text.iter_mut().enumerate() {
+            let mut tab_pos = vec![];
+            for (pos, c) in line.iter().enumerate() {
                 if *c == '\t' {
-                    *c = ' ';
+                    tab_pos.push(pos);
+                }
+            }
+            // start with the tabs at the end of the line to replace them with 4 space chars
```

Comment:
Am I missing something...? Why is it useful/important to go in reverse? (Doesn't seem harmful, but also not imp't.)

Reply:
oh, I see, duh. Otherwise the indices are invalid.